corehq/apps/es/client.py

```diff
@@ -391,6 +401,9 @@ def reindex(self, source, dest, wait_for_completion=False, refresh=False):
             },
             "conflicts": "proceed"
         }
+
+        if purge_ids:
+            reindex_body["script"] = {"inline": "if (ctx._source._id) {ctx._source.remove('_id')}"}
```

Comment:
Is it possible that `ctx._source._id` is empty / null? I assume not, otherwise you could use `ctx._source.hasProperty('_id')` (assuming this is a Groovy script)

Reply:
It is possible, and true for most docs, but the command handles this fine. I belive `ctx._source._id` returns `null`, acting like python's `get`, when `_id` does not exist.